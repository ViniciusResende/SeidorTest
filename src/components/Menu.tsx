import React, {useContext} from 'react'
import { AppContext } from '../contexts/AppContext';
import { Wrapper } from '../styles/Menu.styles';

const Menu: React.FC = () => {
  const {
    isListInDisplay,
    isRegisterInDisplay,
    menuListHandler,
    menuRegisterHandler,
  } = useContext(AppContext);
  return (
    <Wrapper>  
      <div className="menuContainer">
        <button onClick={menuListHandler} className={isListInDisplay ? 'selected' : ''}>Lista</button>
        <button onClick={menuRegisterHandler} className={isRegisterInDisplay ? 'selected' : ''}>Cadastro</button>
      </div>
    </Wrapper>
  )
}

export default Menu
