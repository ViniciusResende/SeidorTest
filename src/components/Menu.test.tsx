import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Menu from './Menu';

const NewMenu = () => {
  return (
    <BrowserRouter>
      <Menu />
    </BrowserRouter>
  )
}

test('Should render the Footer with navigation anchors', () => {
  const tree = renderer
    .create(<NewMenu />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});