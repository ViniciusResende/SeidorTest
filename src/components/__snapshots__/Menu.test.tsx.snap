// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Footer with navigation anchors 1`] = `
<nav
  className="sc-bdfBwQ xUcrl"
>
  <div
    className="sc-gsTCUz ePrpVt"
  >
    <a
      className="sc-dlfnbm gNdSov"
      href="/"
      onClick={[Function]}
    >
      Lista
    </a>
    <a
      className="sc-dlfnbm gNdSov"
      href="/register"
      onClick={[Function]}
    >
      Cadastro
    </a>
    <a
      className="sc-dlfnbm gNdSov"
      href="/update"
      onClick={[Function]}
    >
      Atualizar
    </a>
  </div>
</nav>
`;
